### ISSUE TITLE
[expo-file-system] copyAsync not working

### ISSUE THREAD

**Issue by fpena** [expo/expo#27021]
### Minimal reproducible example

https://snack.expo.dev/@fpena/blissful-green-candies?platform=ios

### Summary

Using the latest Expo version (50.0.0) copying an asset doesn't work with expo-file-system package. 

### Environment

expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.2
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.18.0 - ~/.nvm/versions/node/v18.18.0/bin/node
      Yarn: 1.22.19 - /usr/local/bin/yarn
      npm: 6.14.16 - /usr/local/opt/node@12/bin/npm
    Managers:
      CocoaPods: 1.14.3 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.2, iOS 17.2, macOS 14.2, tvOS 17.2, visionOS 1.0, watchOS 10.2
    IDEs:
      Android Studio: 2022.3 AI-223.8836.35.2231.10671973
      Xcode: 15.2/15C500b - /usr/bin/xcodebuild
    npmPackages:
      expo: ~50.0.6 => 50.0.6
      react: 18.2.0 => 18.2.0
      react-native: 0.73.4 => 0.73.4
    npmGlobalPackages:
      eas-cli: 7.1.2
    Expo Workflow: managed

**Comment by protetore** [comment]
Same here.

The code was working before upgrading to expo 50. I copy a picture selected from camera roll to a temporary dir. It works on Android but fails on iOS.

```
Console Warning
Possible unhandled promise rejection (id: 0):
Error: Calling the 'copyAsync' function has failed
- Caused by: File '/LO/001' is not readable
```

**Comment by protetore** [comment]
Same here!

I'm using `copyAsync` to copy a picture selected from camera roll to a temporary directory. It works on Android, but fails on iOS. It was working fine before updating to Expo 50.

```
Console Warning
Possible unhandled promise rejection (id: 0):
Error: Calling the 'copyAsync' function has failed
- Caused by: File '/LO/001' is not readable
```

**Comment by protetore** [comment]
I think there is still an issue with `copyAsync`, as I mentioned in the closed PR #27208, I managed to test the new version, but I'm facing another problem now: the files gets copied successfully, but the promise never resolves. 

This can be replicated by selecting a photo from camera roll and copying it to the documents path. The promise never resolves, but when a got to the directory and check if the file is there, it is.

```
await FileSystem.copyAsync({
  from: selectedPhoto,
  to: `${fullPath}/${pictureId}`,
});

console.log("You will never see this message");
```

**Comment by fpena** [comment]
> I think there is still an issue with `copyAsync`, as I mentioned in the closed PR #27208, I managed to test the new version, but I'm facing another problem now: the files gets copied successfully, but the promise never resolves.
> 
> This can be replicated by selecting a photo from camera roll and copying it to the documents path. The promise never resolves, but when a got to the directory and check if the file is there, it is.
> 
> ```
> await FileSystem.copyAsync({
>   from: selectedPhoto,
>   to: `${fullPath}/${pictureId}`,
> });
> 
> console.log("You will never see this message");
> ```
> 
> Should I open another issue?

Yes, I think you should. Let me know if you need help with replicating the project or extra assistance.
